zcash_client_sqlite/src/wallet.rs

```diff
@@ -157,29 +163,89 @@ pub(crate) fn add_account<P: consensus::Parameters>(
     params: &P,
     account: AccountId,
     key: &UnifiedFullViewingKey,
+    birthday: Option<AccountBirthday>,
 ) -> Result<(), SqliteClientError> {
-    add_account_internal(conn, params, "accounts", account, key)
-}
+    // Set the wallet birthday, falling back to the chain tip if not specified
+    let chain_tip = scan_queue_extrema(conn)?.map(|(h, _)| h);
+
+    let birthday_height = birthday.as_ref().map_or_else(
+        || chain_tip.ok_or(SqliteClientError::ChainHeightUnknown),
```

Comment:
I'm not certain whether this should be an error or whether we should default to Sapling activation. Strictly speaking it's not an error to create an account in a state where the chain tip is unknown and no wallet birthday is specified.